cli/lib/kontena/cli/log_formatters/compact.rb

```diff
@@ -0,0 +1,55 @@
+require 'logger'
+
+module Kontena
+  module Cli
+    module LogFormatter
+      class Compact < Logger::Formatter
+        def self.ms_since_first
+          ((Time.now.to_f - @first_log) * 1000).to_i
+        end
+
+        def self.ms_since_last
+          ((Time.now.to_f - @last_log) * 1000).to_i
+        ensure
+          @last_log = Time.now.to_f
+        end
+
+        def self.__init_timers__
+          @first_log = Time.now.to_f
+          @last_log = @first_log
+        end
+
+        __init_timers__
+
+        def colorize_severity(severity)
+          case severity[0..0]
+          when 'D' then Kontena.pastel.blue('D')
+          when 'W' then Kontena.pastel.yellow('W')
+          when 'I' then Kontena.pastel.cyan('I')
+          when 'E', 'F' then Kontena.pastel.red('E')
+          else severity[0..0]
+          end
+        end
+
+        def colorize_time
+          elapsed = self.class.ms_since_last
+          str = sprintf("%6s", "#{elapsed}ms")
```

Comment:
12 characters is a bit much. You could then just as well display "HH:MM:SS.%3N" which is 12 chars also.